Dem volume solid objects (#1169)
chaos-polymtl/lethe: doc/source/parameters/dem/solid_objects.rst

=============================
Solid Objects
=============================

> Solid objects are finite auxiliary objects that can be stationary or in motion. Rotating impellers, sliding surfaces, and finite stoppers are examples of solid objects. The main differences between them and floating walls are:

Reviewer: ```suggestion
Solid objects are finite auxiliary objects that can be stationary or in motion. Rotating impellers, sliding surfaces, and finite stoppers are examples of solid objects. The main differences between them and floating walls are:
```

Again the change to virtual wall was not made here?

> 1. floating wall is a plan, while a solid object can be either a volume or a surface

Reviewer 2: ```suggestion
1. floating wall is a plane, while a solid object can be either a volume or a surface
```

.. code-block:: text

 subsection solid objects
   subsection solid surfaces
     # Total number of solid surfaces
     set number of solids  = 1
     subsection solid object 0
       subsection mesh
         set type                   = gmsh
         set file name              = none
         set simplex                = true
         set initial refinement     = 0
         set initial rotation axis  = 1, 0, 0
         set initial rotation angle = 0
         set initial translation    = 0, 0, 0
       end
    
       subsection translational velocity
         set Function expression = 0 ; 0 ; 0
       end

       subsection angular velocity
         set Function expression = 0 ; 0 ; 0
       end

       set center of rotation    = 0., 0., 0.
       set output solid object = true
     end
   end
 end

* The ``number of solids`` parameter defines the total number of solid surfaces used during the simulation.

> * For each solid object, we need a separate subsections (for instance 	``subsection solid object 0``) in which the information of the solid surface 0 (``type``, ``file name``, ``initial refinement``, ``initial translation``, ``initial rotation axis``, ``initial rotation angle``  ) is defined. Each component of the ``initial translation`` and of the ``initial rotation axis`` parameters represent the ``x``, ``y`` and ``z`` axis. The rotation is apply before the translation.

Reviewer 2: ```suggestion
* For each solid object, we need a separate subsections (for instance 	``subsection solid object 0``) in which the information of the solid surface 0 (``type``, ``file name``, ``initial refinement``, ``initial translation``, ``initial rotation axis``, ``initial rotation angle``) is defined. Each component of the ``initial translation`` and of the ``initial rotation axis`` parameters represent the ``x``, ``y`` and ``z`` axes. The rotation is applied before the translation.
```

=============================
Solid Objects
=============================

> Floating meshes (solid objects) are finite (limited) auxiliary objects that can be stationary or moving. Rotating impellers, sliding surfaces, and finite stoppers are examples of floating meshes. The main differences between floating meshes and floating walls are:

Reviewer 3: ```suggestion
Solid objects are finite auxiliary objects that can be stationary or in motion. Rotating impellers, sliding surfaces, and finite stoppers are examples of solid objects. The main differences between them and floating walls are:
```

Reviewer 3: Also change the name below

Reviewer 3: ```
1. floating wall is a plan, while a solid object can be either a volume or a surface
2. floating wall is infinite, while solid object is finite
3. floating wall is stationary while solid object may be stationary or moving.
```

.. note:: 

>     At the moment, solid objects (floating meshes) in Lethe have to be defined using triangular (simplex) meshes. Only triangular 2D meshes of 3D surfaces in 3D DEM simulations are presently supported. Quadrilateral 2D meshes of 3D surfaces and 1D mesh of 2D surfaces are not supported at the moment.

Reviewer 3: ```suggestion
    At the moment, solid objects in Lethe have to be defined using triangular (simplex) meshes. Only triangular 2D meshes of 3D surfaces in simulations are currently supported, meaning that any other meshes will not work.
```
